src/btree/bt_sync.c

```diff
@@ -300,12 +352,27 @@ __sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
 				continue;
 			}
 
+			track_ckpt_progress =
+			    conn->ckpt_verb_start_time.tv_sec > 0;
+
 			if (WT_PAGE_IS_INTERNAL(page)) {
 				internal_bytes += page->memory_footprint;
 				++internal_pages;
+
+				/* Update the checkpoint progress data */
+				if (track_ckpt_progress) {
+					__sync_checkpoint_update_internal(
+					    conn, page->memory_footprint);
+				}
 			} else {
 				leaf_bytes += page->memory_footprint;
 				++leaf_pages;
+
+				/* Update the checkpoint progress data */
+				if (track_ckpt_progress) {
+					__sync_checkpoint_update_leaf(
```

Comment:
both __sync_checkpoint_update_internal and __sync_checkpoint_update_leaf do very similar thing, the size argument is actually the same for both. I wonder if we can replace two functions with a single function or even a macro.

Reply:
But the two functions are operating on two different variables. So I would leave that with 2 inline functions.